### PR TITLE
Modernization: migrate bundle HTML output window from WebView to WKWebView + SwiftUI  #infra

### PR DESCRIPTION
**Pull request by Kaspik**
## Summary

Replaces the deprecated legacy WebKit (`WebView`) bundle HTML output window with a WKWebView-based SwiftUI rewrite, removing all `WebView`/`WebPreferences` deprecation warnings from `SPBundleHTMLOutputController` by deleting it.

Three review-friendly commits:

### Phase 1 — foundation
- **`SAWebView`** — reusable `NSViewRepresentable` WKWebView wrapper (intended for the future help-viewer/tooltip/print migrations): legacy navigation policy (`sequelace://`, `sp-reveal-file://`, `sp-open-file://`), JS alert/confirm/prompt panels, page zoom + pinch zoom, context-menu and key-event hooks.
- **`SABundleJSBridge`** — recreates the legacy `window.system` JS API. Value-returning calls (`run`, `getShellEnvironmentForName`) tunnel through synchronous `window.prompt()` so existing third-party bundles keep working **unmodified**; fire-and-forget calls use a script message handler held via a weak proxy (no retain cycle). JS exceptions surface through an injected `window.onerror` (replaces the private script-debug-delegate API).
- 16 unit tests covering both bridge channels, edge cases (null UUID, malformed payloads) and the injected script.

### Phase 2 — the screen
- **`SABundleHTMLOutputWindowController`** + **`SABundleHTMLOutputView`**, following the established `SAAboutWindowController` hosting pattern (`@objc` NSWindowController + `NSHostingView`).
- Full behavioral port: `displayHTMLContent:withOptions:` / `displayURLString:withOptions:` incl. socket-help/stale-bookmarks frame save & restore, ⌘± zoom and ⌘←/→ history keys, View Source / Save Page As / Print context menu, `SPBundleManager` registration lifecycle.
- Printing via `WKWebView.printOperation(with:)` with SPPrintUtility's margin logic; print-background honors the existing `PrintBackground` default (macOS 13.3+).
- All call sites switched, including the class-name **string** comparisons in `SPCustomQuery`, `SPDatabaseDocument` and `SPTableContent`.

### Phase 3 — cleanup
- Deletes `SPBundleHTMLOutputController.{h,m}` and `BundleHTMLOutput.xib`; drops the dead `WebPreferences` ivar from `SPTooltip.h`.

## Behavior changes
- JS `window.open()` returns no window handle (the window still opens via a sibling controller) — WebKit's create-web-view callback can't synchronously return a SwiftUI-hosted web view.
- The print panel no longer has the live "print background" accessory checkbox; the existing `PrintBackground` preference is applied at print time instead (WKWebView limitation below macOS 13.3).

## Test plan
- [x] App + Unit Tests targets build clean (no new warnings)
- [x] `SABundleJSBridgeTests` (16 tests) pass
- [x] Full unit suite passes (one pre-existing local-only failure: `SPTableContentColumnFilterTests.testAutoFillHeuristicControllerDisabledByDefault` — `PreferenceDefaults.plist` was never in the standalone test bundle; green in CI)
- [ ] Manual: run an HTML-output bundle (e.g. DB Report), socket-help window frame restore, View Source / Save Page As / Print, `window.system.run()` from a test bundle

## Follow-ups
`SPHelpViewerController`, `SPTooltip` leftovers, and the `SPPrintUtility`/`SPPrintAccessory` print pipeline still use legacy WebView — natural next migrations, now able to reuse `SAWebView`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Replaced the legacy HTML output window with a modern WebView-based experience featuring improved pinch + shortcut zoom, back/forward navigation, and preserved legacy JavaScript behavior.
  * Added/strengthened HTML output actions and context menu items: View Source, Save Page As…, and Print Page.
* **Bug Fixes**
  * Improved HTML output window lifecycle, including reliable handling for opening links in new windows and preventing duplicate output windows for the same session.
* **Tests**
  * Added unit tests for the JavaScript bridge injection and message handling.

### CODE REVIEW

**Comment by Kaspik** [comment]
@codex review

**Comment by Jason-Morcos** [comment]
@codex review

File: Source/Controllers/BundleSupport/SABundleHTMLOutputWindowController.swift [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+    func windowWillClose(_ notification: Notification) {
+        if restoreFrame {
+            window?.setFrame(origFrame, display: true)
+            restoreFrame = false
+        }
+        if windowType == SPConnectionShownSocketHelp {
+            windowType = nil
+            // Set once they close the window, matching the legacy behavior.
+            UserDefaults.standard.set(true, forKey: SPConnectionShownSocketHelp)
+        }
```
Comment (Jason-Morcos) [inline Source/Controllers/BundleSupport/SABundleHTMLOutputWindowController.swift]: This seems like a separation of concerns smell - maybe this should belong wherever triggers off the socket help window in the first place?

**Review by Jason-Morcos — APPROVED** [review]
Cool! Assuming you've tested this extensively locally, I'm good with it
Glad to see an xib go!

File: Source/Controllers/BundleSupport/SABundleHTMLOutputWindowController.swift (hunk 1 above)
Comment (Kaspik) [reply Source/Controllers/BundleSupport/SABundleHTMLOutputWindowController.swift]: Good call — fixed in 7899a166b. The controller no longer knows anything about socket-help: it now treats the `"frame"` display option generically (resize, restore on close) and exposes a generic `windowWillCloseHandler` hook instead of checking `SPConnectionShownSocketHelp`/`SPStaleBookmarksHelp` and writing the default itself. `SPConnectionController`, which opens the socket-help window, now sets that default via the close handler. No behavior change — it was the only caller passing those options (and `SPStaleBookmarksHelp` had no caller at all).

**Comment by Jason-Morcos** [comment]
@codex review
